torchvision/datasets/hmdb51.py

```diff
@@ -78,38 +81,50 @@ def __init__(self, root, annotation_path, frames_per_clip, step_between_clips=1,
             _video_min_dimension=_video_min_dimension,
             _audio_samples=_audio_samples,
         )
+        self.fold = fold
+        self.train = train
+        self.classes = classes
         self.video_clips_metadata = video_clips.metadata
-        self.indices = self._select_fold(video_list, annotation_path, fold, train)
+        self.indices = self._select_fold(video_paths, annotation_path, fold, train)
         self.video_clips = video_clips.subset(self.indices)
         self.transform = transform
 
     @property
     def metadata(self):
         return self.video_clips_metadata
 
-    def _select_fold(self, video_list, annotation_path, fold, train):
-        target_tag = 1 if train else 2
-        name = "*test_split{}.txt".format(fold)
-        files = glob.glob(os.path.join(annotation_path, name))
+    def _select_fold(self, video_list, annotations_dir, fold, train):
+        target_tag = self.TRAIN_TAG if train else self.TEST_TAG
+        split_pattern_name = "*test_split{}.txt".format(fold)
+        split_pattern_path = os.path.join(annotations_dir, split_pattern_name)
+        annotation_paths = glob.glob(split_pattern_path)
         selected_files = []
-        for f in files:
-            with open(f, "r") as fid:
-                data = fid.readlines()
-                data = [x.strip().split(" ") for x in data]
-                data = [x[0] for x in data if int(x[1]) == target_tag]
-                selected_files.extend(data)
+        for filepath in annotation_paths:
+            with open(filepath) as fid:
+                lines = fid.readlines()
+            for line in lines:
+                video_filename, tag_string = line.split()
+                tag = int(tag_string)
+                if tag == target_tag:
+                    selected_files.append(video_filename)
         selected_files = set(selected_files)
-        indices = [i for i in range(len(video_list)) if os.path.basename(video_list[i]) in selected_files]
+
+        indices = []
+        for video_index, video_path in enumerate(video_list):
+            if os.path.basename(video_path) in selected_files:
+                indices.append(video_index)
```

Comment:
I think this is personal taste, but list comprehensions are pretty standard in pythonic code (but maybe it wasn't very clear in my code before).

Reply:
Sure! I like list comprehensions too. But
1. I think this is more readable (at least I would understand it faster if I had to read it for the first time)
2. It's more pythonic to iterate over a sequence of items `for a in b: use(a)` than doing `for i in range(len(b)): use(a[i])`.